packages/ng-query/src/lib/tests/entity-utils.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { TestBed } from '@angular/core/testing';
+import { QueryClientService } from '../query-client';
+import { arrayPush } from '../entity-utils';
+
+interface Item {
+  id: string;
+}
+
+describe('entity utils', () => {
+  it('should add entity to the query', () => {
+    const { client } = setup();
+    const item1: Item = { id: 'item-1' };
+    const item2: Item = { id: 'item-2' };
+
+    client.setQueryData(['item'], item1);
+    client.setQueryData(['item'], arrayPush('item', item2));
+
+    const items = client.getQueryData<Item[]>(['item']);
+
+    expect(items?.length).toBe(2);
```

Comment:
@NetanelBasal the test fail for existing method. I believe `[state` ](https://github.com/ngneat/query/blob/dfc9e81a3efb49e36e34ab6df69bf5e00b3d339a/packages/ng-query/src/lib/entity-utils.ts#L8)provided by query client is not dictionary of keys-data, instead its just previous data, that's why it throws error. Please have a look.

Reply:
The state should be an object, and the key you want to modify should be an array. 

```ts

const items = client.getQueryData<{ items: Item[] }>(['items']);

client.setQueryData(['item'], arrayPush('items', item2));
```